src/tangent_types/abstract_zero.jl

```diff
@@ -87,3 +87,32 @@ arguments.
 ```
 """
 struct NoTangent <: AbstractZero end
+
+"""
+    zero_tangent(primal)
+
+This returns an appropriate zero tangent suitable for accumulating tangents of the primal.
+For mutable composites types this is a structural []`MutableTangent`](@ref)
+For `Array`s, it is applied recursively for each element.
+For immutable types, this is simply [`ZeroTangent()`](@ref) as accumulation is default out-of-place for contexts where mutation does not apply.
+(Where mutation is not to be supported even for mutable types, then [`ZeroTangent()`](@ref) should be used for everything)
+
+!!! warning Exprimental
+    `zero_tangent`is an experimental feature, and is part of the mutation support featureset.
+    While this notice remains it may have changes in behavour, and interface in any _minor_ version of ChainRulesCore.
+    Exactly how it should be used (e.g. is it forward-mode only?)
+"""
+function zero_tangent end
+zero_tangent(::AbstractString) = ZeroTangent()
```

Comment:
Should this be `zero_tangent` or `NoTangent`?

Reply:
`NoTangent` is always correct for a string.
`ZeroTangent` is only correct for a string if it is somewhere that it is known the tangent should be zero -- which is what `zero_tangent` is intended for.
So idk either works.

Reply:
Fair enough